### DISCUSSION TITLE
New User: Unable to Push File

### DISCUSSION THREAD

**Discussion by CodezHere** [community/community#22772] [New to GitHub]
<p>Hello Gitworld,</p>
<p>I’m a new user to both Ubuntu Linux (WSL) and Github, so bear with me.</p>
<p>So far I’ve setup my account, added my SSH key and I’ve authenitcated in my WSL terminal.</p>
<p>I wanted to try out pushing and pulling, so I created a new directory and created a new Git instance locally, then created a basic readme.txt within said directory. I created a repository on GitHub with the same name as the local directory (instance).</p>
<p>Now the issue… I connected my local repository to the remote one:</p>
<blockquote>
<p>git remote add origin </p>
</blockquote>
<p>but when I push:</p>
<blockquote>
<p>git push origin master</p>
</blockquote>
<p>I recieve this error message:</p>
<blockquote>
<p>git: ‘remote-https’ is not a git command. See ‘git --help’.</p>
</blockquote>
<p>I’m not sure what the error is, since I’m not using a ‘remote-http’ command at all in the line?</p>
<p>Any guidance would be much appreciated, thank you in advance.</p>

**Comment by kingthorin** [comment]
<p>Sounds like you made a mistake when you did your <code>git remote add origin &lt;git_uri&gt;</code></p>
<p>Try <code>git remote -v</code> and see if it is as expected.</p>

**Comment by CodezHere** [comment]
<p>when I try git remove -v it gives:</p>
<blockquote>
<p>git: ‘remove’ is not a git command.</p>
</blockquote>

**Comment by kingthorin** [comment]
<p>“remote” with a T :wink:</p>

**Comment by CodezHere** [comment]
<p>oh jeez… thank you.<br>
output:<br>
origin  <a href="https://github.com/" rel="noopener nofollow ugc">https://github.com/</a>/heelo-world.git (fetch)<br>
origin  <a href="https://github.com/" rel="noopener nofollow ugc">https://github.com/</a>/heelo-world.git (push)</p>

**Comment by CodezHere** [comment]
<p>I removed my username between the URL and .git file, but both were identical. Is that to be expected?</p>

**Comment by airtower-luna** [comment]
<p>Yes, in most cases push and pull URLs for a remote are the same. However you mentioned wanting to use SSH authentication. To do that you need to change the URL to the SSH URL.</p>
<p>You can find the URL using the green “Code” button on your repository page and selecting SSH:<br>
<img src="https://user-images.githubusercontent.com/5814919/181073090-493be8ac-3d8f-4a5c-a4ca-66cb67937f23.png" alt="grafik" data-base62-sha1="ArrILo12yKZq1EUNWoJiDLpGjhv" width="363" height="181"></p>
<p>Then use <code>git remote set-url origin SSH_URL</code> to update the URL.</p>

**Comment by kingthorin** [comment]
<blockquote>
<p>I removed my username between the URL and .git file, but both were identical. Is that to be expected?</p>
</blockquote>
<p>Yes, but not if you plan to use SSH.</p>
<p>You should <code>git remote remove origin</code> then <code>git remote add origin git@github.com:&lt;owner&gt;/&lt;repo&gt;.git</code>. (<a class="mention" href="/airtower-luna">@airtower-luna</a> pointed out where to copy the SSH URL)</p>

**Comment by airtower-luna** [comment]
<p>There’s no need to remove the remote though, you can just change the URL. :wink:</p>

**Comment by kingthorin** [comment]
<p>You mean if you just do an add with a colliding name it’ll replace the URL? I’ve never tried.</p>

**Comment by airtower-luna** [comment]
<p>No, you can change the URL of the existing remote:</p>
<pre><code class="lang-bash">git remote set-url origin URL_GOES_HERE
</code></pre>
<p>There are also options to change e.g. just the push URL, but I don’t see a need for that here.</p>

**Comment by kingthorin** [comment]
<p>Oh okay, I see what you’re getting at. Thanks!</p>

**Comment by CodezHere** [comment, marked as answer]
<p>Thank you <a class="mention" href="/kingthorin">@kingthorin</a> &amp;  <a class="mention" href="/airtower-luna">@airtower-luna</a>! Appreciate all the replies and help. I had a feeling I had to use the SSH URL, but wasn’t sure. I found a solution in the end by switching to it, but everytime I used:</p>
<p>git push origin master</p>
<p>I would get the error:</p>
<p>error: src refspec master does not match any<br>
error: failed to push some refs to ‘<a href="http://github.com" rel="noopener nofollow ugc">github.com</a>:CodezHere/heelo-world.git’</p>
<p>When I used:</p>
<p>git push origin</p>
<p>It worked!<br>
What is the difference between the two? Am I losing anything by not being able to run the “git push origin master” command?</p>

**Comment by airtower-luna** [comment]
<p>The difference is with <code>git push origin master</code> you explicitly tell git to push the <code>master</code> branch to a matching remote ref, without it some defaults apply. I can’t guess why that form didn’t work, there are a lot of configuration option that affect what exactly might get pushed how, see the <a href="https://git-scm.com/docs/git-push"><code>git push</code> documentation</a>. Common use is to <a href="https://git-scm.com/docs/git-push#Documentation/git-push.txt---set-upstream">set an upstream branch</a> for branches you commonly push so git knows what to do on <code>git push</code> without any options.</p>
